samples/js/node/pronunciationAssessment.js

```diff
@@ -0,0 +1,295 @@
+    var durations = [];
+    var jo = {};
+        
+    // Before beginning speech recognition, setup the callbacks to be invoked when an event occurs.
+
+    // The event recognizing signals that an intermediate recognition result is received.
+    // You will receive one or more recognizing events as a speech phrase is recognized, with each containing
+    // more recognized speech. The event will contain the text for the recognition since the last phrase was recognized.
+    reco.recognizing = function (s, e) {};
+
+    // The event recognized signals that a final recognition result is received.
+    // This is the final event that a phrase has been recognized.
+    // For continuous recognition, you will get one recognized event for each phrase recognized.
+    reco.recognized = function (s, e) {
+        console.log("pronunciation assessment for: ", e.result.text);
+        var pronunciation_result = sdk.PronunciationAssessmentResult.fromResult(e.result);
+        console.log(" Accuracy score: ", pronunciation_result.accuracyScore, '\n',
+            "pronunciation score: ", pronunciation_result.pronunciationScore, '\n',
+            "completeness score : ", pronunciation_result.completenessScore, '\n',
+            "fluency score: ", pronunciation_result.fluencyScore
+        );
+
+        jo = eval("(" + e.result.properties.getProperty(sdk.PropertyId.SpeechServiceResponse_JsonResult) + ")");
+        const nb = jo["NBest"][0];
+        startOffset = nb.Words[0].Offset;
+        const localtext = _.map(nb.Words, (item) => item.Word.toLowerCase());
+        currentText = currentText.concat(localtext);
+        fluencyScores.push(nb.PronunciationAssessment.FluencyScore);
+        const isSucceeded = jo.RecognitionStatus === 'Success';
+        const nBestWords = jo.NBest[0].Words;
+        const durationList = [];
+        _.forEach(nBestWords, (word) => {
+            recognizedWords.push(word);
+            durationList.push(word.Duration);
+        });
+        durations.push(_.sum(durationList));
+
+        if (isSucceeded && nBestWords) {
+            allWords.push(...nBestWords);
+        }
+
+        const resText = currentText.join(" ");        
+        let wholelyricsArry = [];
+        let resTextArray = [];
+        if (["zh-cn"].includes(settings.language.toLowerCase())) {
+            const resTextProcessed = (resText.toLocaleLowerCase() ?? "").replace(new RegExp("[^a-zA-Z0-9\u4E00-\u9FA5']+", "g"), " ");
+            const wholelyrics = (reference_text.toLocaleLowerCase() ?? "").replace(new RegExp("[^a-zA-Z0-9\u4E00-\u9FA5']+", "g"), " ");
+            const segment = new Segment();
+            segment.useDefault();
+            segment.loadDict('wildcard.txt');
+            _.map(segment.doSegment(wholelyrics, {stripPunctuation: true}), (res) => wholelyricsArry.push(res['w']));
+            _.map(segment.doSegment(resTextProcessed, {stripPunctuation: true}), (res) => resTextArray.push(res['w']));
+        } else {
```

Comment:
how about other non-English locales?

Reply:
We just provide `zh-CN` word segmentation sample and for other locales, we just remove the punctuation like what we have done in the Python sample code.

Reply:
Does the service only support `zh-cn` and `en-us`? If not, please update readme to clearify the limitation of the sample codes

Reply:
No. Support all locale. But for other locales which need word segmentation, the segmentation should be implemented by users. For other non-English locales, the non-word characters (e.g, punctuation) should be removed by users. We can not provide samples for all locales.
Anyway, @l17813182030 you can add some comments here to notice the users.

Reply:
Is there some library could be leveraged?

Reply:
> No. Support all locale. But for other locales which need word segmentation, the segmentation should be implemented by users. For other non-English locales, the non-word characters (e.g, punctuation) should be removed by users. We can not provide samples for all locales. Anyway, @l17813182030 you can add some comments here to notice the users.

have added comments to notice the users

Reply:
> Is there some library could be leveraged?

No similar library is found in js